src/sp/it/pl/audio/Item.kt

```diff
@@ -24,9 +24,9 @@ abstract class Item {
 
     /**
      * @return absolute file this item represents
-     * @throws RuntimeException if this item is not file based
+     * @throws IllegalArgumentException if this item is not file based
```

Comment:
There is no reason to narrow down exception type. Pls use RuntimeException. Note that in the future, the method will return null or Try instead.

Reply:
IllegalArgumentException is the exception that makes most sense here, since a non-file-based URI is an illegal argument to this function

Reply:
Types of exceptions make no sense if noone is going/supposed to catch them, we are merely crashing the program here. This is an assert. Giving any specifics on the exception is meaningless here, in fact only distracts the developer. The throws clause was left there to document the invocation conditions the method relies on. Documenting runtime exceptions at all is counter-intuitive and wrong.

Pls, change the javadoc to describe the conditions and remove the throws altogether. There may be additional such methods in this class. Every one should document it the same way.

Reply:
so it should return null if it fails? But then you could just straight use `toFileOrNull()`, this function would be unneccessary in that case

Reply:
No it should fail. The conditions mandating the failure should be documented (item must be file based), the exception must not.